libs/sdk-common/src/input_parser.rs

```diff
@@ -188,6 +186,15 @@ const BIP353_PREFIX: &str = "bitcoin:";
 pub async fn parse(
     input: &str,
     external_input_parsers: Option<&[ExternalInputParser]>,
+) -> Result<InputType> {
+    let rest_client: Arc<dyn RestClient> = Arc::new(ReqwestRestClient::new()?);
+    parse_with_rest_client(rest_client, input, external_input_parsers).await
+}
+
+pub async fn parse_with_rest_client(
+    rest_client: Arc<dyn RestClient>,
```

Comment:
The docs for the reqwest client say
> You do not have to wrap the Client in an [Rc] or [Arc](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) to reuse it, because it already uses an [Arc](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) internally.

So nit: I think functions can use a borrowed rest client rather than an Arc. That saves clones on passing it to the next function.

Reply:
Is this what you had in mind? [0f4633d](https://github.com/breez/breez-sdk-greenlight/pull/1175/commits/0f4633d245bdc594347cb9b3f9304955bab4104e)

Reply:
Not really. I had in mind 
```
pub async fn validate_lnurl_withdraw(rest_client: &dyn RestClient)

... 

let mock_rest_client = MockRestClient::new();
validate_lnurl_withdraw(&mock_rest_client, withdraw_req, invoice);
```

But coding it up it would require generics, to support cloning the rest client. It would get quite some feet in the earth. So let's stick with the arcs.